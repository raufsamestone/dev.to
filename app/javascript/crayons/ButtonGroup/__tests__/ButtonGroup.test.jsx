import { h } from 'preact';
import render from 'preact-render-to-json';
import {
  Button,
  ButtonGroup,
  OutlinedButton,
  DangerButton,
  SecondaryButton,
} from '@crayons';

describe('<ButtonGroup /> component', () => {
  it('should render', () => {
    const Icon = () => (
      <svg
        width="24"
        height="24"
        xmlns="http://www.w3.org/2000/svg"
        className="crayons-icon"
      >
        <path d="M9.99999 15.172L19.192 5.979L20.607 7.393L9.99999 18L3.63599 11.636L5.04999 10.222L9.99999 15.172Z" />
      </svg>
    );

    const tree = render(
      <ButtonGroup>
        {' '}
        <Button>I&apos;m the primary</Button>
        <OutlinedButton>I&apos;m outlined</OutlinedButton>
        <DangerButton>Danger!</DangerButton>
        <SecondaryButton icon={Icon}>Hello!</SecondaryButton>
        <SecondaryButton icon={Icon} />
      </ButtonGroup>,
    );
    expect(tree).toMatchSnapshot();
  });
});
