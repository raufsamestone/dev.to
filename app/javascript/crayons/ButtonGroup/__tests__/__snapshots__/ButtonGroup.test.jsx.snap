// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ButtonGroup /> component should render 1`] = `
<div
  class="crayons-btn-group"
>
   
  <button
    class="crayons-btn"
    disabled={false}
    type="button"
  >
    I'm the primary
  </button>
  <button
    class="crayons-btn crayons-btn--outlined"
    disabled={false}
    type="button"
  >
    I'm outlined
  </button>
  <button
    class="crayons-btn crayons-btn--danger"
    disabled={false}
    type="button"
  >
    Danger!
  </button>
  <button
    class="crayons-btn crayons-btn--secondary crayons-btn--icon-left"
    disabled={false}
    type="button"
  >
    <svg
      class="crayons-icon"
      height="24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M9.99999 15.172L19.192 5.979L20.607 7.393L9.99999 18L3.63599 11.636L5.04999 10.222L9.99999 15.172Z"
      />
    </svg>
    Hello!
  </button>
  <button
    class="crayons-btn crayons-btn--secondary crayons-btn--icon-alone"
    disabled={false}
    type="button"
  >
    <svg
      class="crayons-icon"
      height="24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M9.99999 15.172L19.192 5.979L20.607 7.393L9.99999 18L3.63599 11.636L5.04999 10.222L9.99999 15.172Z"
      />
    </svg>
  </button>
</div>
`;
